src/questions/owner_borrow.yaml

```diff
@@ -240,3 +240,751 @@
+    - content: "revert, opened"
+      key: c
+    - content: "scope, opened"
+      key: D
+
+- uuid: e539b827-cbf2-4b35-8fcc-af16c8d0c375
+  type: SingleChoice
+  content: which statement best desribe reference?
+  hint: pointer
+  explanation: "point to valid value"
+  answerKeys:
+    - B
+  subTopics:
+    - borrowing
+  difficultyLevel: low
+  choices:
+    - content: "a set of rules that governs how a Rust program manages memory"
+      key: A
+    - content: "a reference is guaranteed to point to a valid value of a particular type for the life of that reference."
+      key: B
+    - content: "new concept for many programmers, it does take some time to get used to"
+      key: c
+    - content: "the range within a program for which an item is valid"
+      key: D
+
+- uuid: 3661c275-8559-40e4-b5dd-ac138e3c5be9
+  type: SingleChoice
+  content: The action of creating a reference to a variable is called ____
+  hint: NoHint
+  explanation: "creation of reference is pointer to a variable"
+  answerKeys:
+    - D
+  subTopics:
+    - borrowing
+  difficultyLevel: low
+  choices:
+    - content: "Referencing"
+      key: A
+    - content: "Ownership"
+      key: B
+    - content: "Heaping"
+      key: c
+    - content: "Borrowing"
+      key: D
+
+- uuid: 434bc0ff-02a9-4287-a469-e74c3b673f70
+  type: SingleChoice
+  content: What is a Scope?
+  hint: noHint
+  explanation: ""
+  answerKeys:
+    - A
+  subTopics:
+    - memory-allocation
+  difficultyLevel: low
+  choices:
+    - content: "the range within a program for which an item is valid"
+      key: A
+    - content: "the range within a program for which an item is invalid"
+      key: B
+    - content: "the range within a program"
+      key: c
+    - content: "None of the above"
+      key: D
+
+- uuid: 896206c4-09f5-4c46-8cfd-6f5597df0c02
+  type: SingleChoice
+  content: When your code calls a function, the values passed into the function are pushed into
+  hint: not a first in first out
+  explanation: "function are pushed in and dont have to leave first"
+  answerKeys:
+    - B
+  subTopics:
+    - memory-allocation
+  difficultyLevel: low
+  choices:
+    - content: "Stack"
+      key: A
+    - content: "Heap"
+      key: B
+    - content: "Memory"
+      key: c
+    - content: "calldata"
+      key: D
+
+- uuid: 762de2f4-3e2f-42a3-ba93-770ab7a833dc
+  type: SingleChoice
+  content: The function's local variables get pushed onto
+  hint: First in First Out
+  explanation: "pushed accordingly"
+  answerKeys:
+    - D
+  subTopics:
+    - memory-allocation
+  difficultyLevel: low
+  choices:
+    - content: "Heap"
+      key: A
+    - content: "Calldata"
+      key: B
+    - content: "Function"
+      key: c
+    - content: "Stack"
+      key: D
+
+- uuid: aa969d59-a963-442d-b13b-d83ee2f65f28
+  type: SingleChoice
+  content: When the function is over, those values get_______ the stack.
+  hint: correct term for memory allocation
+  explanation: "The right term is pop out"
```

Comment:
We can explain it in a better and detail way

----

Well, when a function gets called, some memory gets allocated for all of its local variables and some other information. This is called a ‘stack frame’, and for the purpose of this tutorial, we’re going to ignore the extra information and only consider the local variables we’re allocating. So in this case, when main() is run, we’ll allocate a single 32-bit integer for our stack frame. This is automatically handled for you, as you can see; we didn’t have to write any special Rust code or anything.

When the function exits, its stack frame gets deallocated. This happens automatically as well.